### PR TITLE
Update the calculation of uvel and vvel in evp dynamics

### LINKED ISSUE #951: Minor modifications required in documentation
- Remove sentence: 'Here we assume that the top layer is thin enough.' in section 2.1
- the eqs in section 2.5 assume that the ice is in the northern hemisphere (f>0) for the water stress term when the turning angle is not zero. Need to mention it or modify the eqs to have a 'sign' variable for some terms (as done in the code).

### LINKED ISSUE #952: Unnecessary calculations for uvel, vvel for the C-grid
In ice_dyn_evp:

call grid_average_X2Y('S', uvelE, 'E', uvel, 'U')
call grid_average_X2Y('S', vvelN, 'N', vvel, 'U')

uvel(:,:,:) = uvel(:,:,:)*uvm(:,:,:)
vvel(:,:,:) = vvel(:,:,:)*uvm(:,:,:)

Let's look at uvel = uU (same idea for vvel). In the grid_average we have:

uvel(i,j) = (uvelE(i,j) AreaE(i,j) MaskE(i,j) + uvelE(i,j+1) AreaE(i,j+1) MaskE(i,j+1)) / (AreaE(i,j) MaskE(i,j) + AreaE(i,j+1) MaskE(i,j+1))

The multiplications by the mask are not necessary. For example, if MaskE(i,j) = 0, the mask at the U point is also zero. and the next operation uvel(:,:,:) = uvel(:,:,:)*uvm(:,:,:) sets uvel to zero. 

we could just have:

uvel(i,j) = (uvelE(i,j) AreaE(i,j) + uvelE(i,j+1) AreaE(i,j+1)) / (AreaE(i,j) + AreaE(i,j+1))
uvel(:,:,:) = uvel(:,:,:)*uvm(:,:,:)

### PR DESCRIPTION
**Pull request by apcraig**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update the calculation of uvel and vvel in evp to reduce flops, bit-for-bit.
- [X] Developer(s): 
    apcraig, jflemiuex, eclare
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Full test suites run on derecho with intel, gnu, cray.  All testing passes as expected, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#c5b390ec0507633c319e6c7d1cb2a6850924dc59
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please document the changes in detail, including _why_ the changes are made.  This will become part of the PR commit log.

Update the calculation of uvel and vvel in subroutine evp in file ice_dyn_evp.F90.  Do an unmasked grid_average_X2Y when averaging from uvelE to uvel and from vvelN to vvel instead of a masked average. The masking is handled separately.  This change is bit-for-bit and saves a few flops.  Closes #952.

Update the documentation to add the hemispheric sign dependence in the water stress terms in the dynamics equations.  Closes #951.

### CODE REVIEW

**Comment by apcraig** [comment]
I'm doing a full test suite to make sure answers haven't changed.  

@JFLemieux73 @eclare108213 Please review code changes and documentation, let me know if I've done something wrong.

File: doc/source/science_guide/sg_dynamics.rst [hunk 1]
```diff
@@ -35,11 +35,11 @@ For clarity, the two components of Equation :eq:`vpmom` are
    \begin{aligned}
    m{\partial u\over\partial t} &= {\partial\sigma_{1j}\over\partial x_j} + \tau_{ax} +
      a_i c_w \rho_w
-     \left|{\bf U}_w - {\bf u}\right| \left[\left(U_w-u\right)\cos\theta - \left(V_w-v\right)\sin\theta\right]
+     \left|{\bf U}_w - {\bf u}\right| \left[ + \left(U_w-u\right)\cos\theta \mp \left(V_w-v\right)\sin\theta\right]
```
Comment (JFLemieux73) [inline doc/source/science_guide/sg_dynamics.rst:38]: not sure I would add the + sign before \left(U_w...

File: doc/source/science_guide/sg_dynamics.rst [hunk 2]
```diff
@@ -121,38 +121,38 @@ variables used in the code.
...
+     &+ {\tt vrel}\underbrace{\left( \pm U_w\sin\theta+V_w\cos\theta\right)}_{\tt watery}  + {m\over\Delta t_e}v^k,
    :label: vmom
 
 where :math:`{\tt vrel}\ \cdot\ {\tt waterx(y)}= {\tt taux(y)}` and the definitions of :math:`u^{l}` and :math:`v^{l}` vary depending on the grid.
 
 As :math:`u` and :math:`v` are collocated on the B grid, :math:`u^{l}` and :math:`v^{l}` are respectively :math:`u^{k+1}` and :math:`v^{k+1}` such that this system of equations can be solved as follows. Define
 
 .. math::
-   \hat{u} = F_u + \tau_{ax} - mg{\partial H_\circ\over\partial x} + {\tt vrel} \left(U_w\cos\theta - V_w\sin\theta\right) + {m\over\Delta t_e}u^k
+   \hat{u} = F_u + \tau_{ax} - mg{\partial H_\circ\over\partial x} + {\tt vrel} \left(+ U_w\cos\theta \mp V_w\sin\theta\right) + {m\over\Delta t_e}u^k
```
Comment (JFLemieux73) [inline doc/source/science_guide/sg_dynamics.rst:143]: same comment here not sure I would the + sign before U_w

**Review by JFLemieux73 — APPROVED** [review]
looks good!

**Comment by apcraig** [comment]
Testing looks good, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#c5b390ec0507633c319e6c7d1cb2a6850924dc59.

File: doc/source/science_guide/sg_dynamics.rst (hunk 2 above)
Comment (apcraig) [reply doc/source/science_guide/sg_dynamics.rst:143]: This was done just to keep the equations aligned a bit in the text.  I'll remove them again, wasn't sure what was better.

Comment (apcraig) [reply doc/source/science_guide/sg_dynamics.rst]: The extra plus signs are out, feel free to have another look.  Can revert if you prefer the prior version.  If I don't hear anything, will merge later today.

**Pull request merged**